internal/queue/queue.go

```diff
@@ -376,6 +377,7 @@ func (q *Queue) handleQueueItem(ctx context.Context) bool {
 	// We do this as the delayed nature of the work Queue means the items in the informer cache may actually be more u
 	// to date that when the item was initially put onto the workqueue.
 	ctx = span.WithField(ctx, "key", qi.key)
+	ctx = log.WithLogger(ctx, log.G(ctx).WithField("key", qi.key))
```

Comment:
Shouldn't the logger already have this key? If not that seems like a bug in the span implementation.

Reply:
ping